clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/KubernetesUtil.groovy

```diff
@@ -48,6 +50,59 @@ class KubernetesUtil {
     rc.spec?.template?.spec?.imagePullSecrets?.collect({ it.name })
   }
 
+  static KubernetesImageDescription buildImageDescription(String image) {
+    def sIndex = image.indexOf('/')
+    def result = new KubernetesImageDescription()
+
+    // No slash means we only provided a repository name & optional tag.
+    if (sIndex < 0) {
+      result.repository = image
+    } else {
+      def sPrefix = image.substring(0, sIndex)
+
+      // Check if the content before the slash is a registry (either localhost, or a URL)
+      if (sPrefix.startsWith('localhost') || sPrefix.contains('.')) {
+        result.registry = sPrefix
+
+        image = image.substring(sIndex + 1)
+      }
+    }
+
+    def cIndex = image.indexOf(':')
+
+    if (cIndex < 0) {
+      result.repository = image
+    } else {
+      result.tag = image.substring(cIndex + 1)
+      result.repository = image.subSequence(0, cIndex)
+    }
+
+    normalizeImageDescription(result)
+    result
+  }
+
+  static Void normalizeImageDescription(KubernetesImageDescription image) {
+    if (!image.registry) {
+      image.registry = DEFAULT_REGISTRY
+    }
+
+    if (!image.tag) {
+      image.tag = "latest"
+    }
+
+    if (!image.repository) {
+      throw new IllegalArgumentException("Image descriptions must provide a repository.")
+    }
+  }
+
+  static String getImageId(KubernetesImageDescription image) {
+    "$image.registry/$image.repository:$image.tag"
```

Comment:
Could also define a `toString()` directly on `KubernetesImageDescription` and then have that delegate to `getImageId()` below. Would let you drop a bunch of explicit calls to this `getImageId(KubernetesImageDescription)`.